integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/saltpassword/RotateSaltPasswordE2ETest.java

```diff
@@ -0,0 +1,139 @@
+package com.sequenceiq.it.cloudbreak.testcase.e2e.saltpassword;
+
+import java.time.LocalDate;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.Test;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackV4Response;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.instancegroup.instancemetadata.InstanceMetaDataV4Response;
+import com.sequenceiq.common.api.type.InstanceGroupType;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status;
+import com.sequenceiq.it.cloudbreak.client.DistroXTestClient;
+import com.sequenceiq.it.cloudbreak.client.FreeIpaTestClient;
+import com.sequenceiq.it.cloudbreak.client.SdxTestClient;
+import com.sequenceiq.it.cloudbreak.context.Description;
+import com.sequenceiq.it.cloudbreak.context.TestContext;
+import com.sequenceiq.it.cloudbreak.dto.distrox.DistroXTestDto;
+import com.sequenceiq.it.cloudbreak.dto.freeipa.FreeIpaTestDto;
+import com.sequenceiq.it.cloudbreak.dto.sdx.SdxInternalTestDto;
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+import com.sequenceiq.it.cloudbreak.testcase.e2e.AbstractE2ETest;
+import com.sequenceiq.it.cloudbreak.util.spot.UseSpotInstances;
+import com.sequenceiq.it.cloudbreak.util.ssh.action.SshSaltPasswordActions;
+import com.sequenceiq.sdx.api.model.SdxClusterStatusResponse;
+
+public class RotateSaltPasswordE2ETest extends AbstractE2ETest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RotateSaltPasswordE2ETest.class);
+
+    private static final LocalDate PAST_DATE = LocalDate.now().minusMonths(1);
+
+    private static final AtomicReference<String> SHADOW_LINE_REFERENCE = new AtomicReference<>();
+
+    @Inject
+    private SshSaltPasswordActions sshSaltPasswordActions;
+
+    @Inject
+    private FreeIpaTestClient freeIpaTestClient;
+
+    @Inject
+    private SdxTestClient sdxTestClient;
+
+    @Inject
+    private DistroXTestClient distroXTestClient;
+
+    @Override
+    protected void setupTest(TestContext testContext) {
+        createDefaultUser(testContext);
+        createDefaultCredential(testContext);
+        initializeDefaultBlueprints(testContext);
+        createDefaultEnvironment(testContext);
+        createFreeipa(testContext);
```

Comment:
You can use the `createEnvironmentWithFreeIpa` or `createDefaultDatalake` from the abstract class.

Reply:
It is just a small optimization to not create the datalake/datahub until the previous test passed

Reply:
So you can apply the `createEnvironmentWithFreeIpa`.